ingest-app/src/cmr/ingest/validation/validation.clj

```diff
@@ -203,19 +200,26 @@
   "Optional validations whose errors will be returned as warnings."
   [context]
   (let [kms-index (kms-fetcher/get-kms-index context)]
-    {:Platforms [(match-kms-keywords-validation
-                  kms-index :platforms msg/platform-not-matches-kms-keywords)
-                 (v/every {:Instruments (match-kms-keywords-validation
-                                         kms-index :instruments
-                                         msg/instrument-not-matches-kms-keywords)})]
-     :ScienceKeywords (match-kms-keywords-validation
-                       kms-index :science-keywords msg/science-keyword-not-matches-kms-keywords)
-     :Projects (match-kms-keywords-validation
-                kms-index :projects msg/project-not-matches-kms-keywords)
-     :LocationKeywords (match-kms-keywords-validation
-                        kms-index :spatial-keywords msg/location-keyword-not-matches-kms-keywords)
-     :DataCenters (match-kms-keywords-validation
-                   kms-index :providers msg/data-center-not-matches-kms-keywords)}))
+    (merge (useconstraints-onlineresource-validators kms-index)
+           (collectioncitations-onlineresource-validators kms-index)
+           (publicationreferences-onlineresource-validators kms-index)
+           (datacenter-url-validators kms-index true)
+           (contactpersons-url-validators kms-index true)
+           (contactgroups-url-validators kms-index true)
+           (related-url-validator kms-index true)
```

Comment:
should we not instead use the conditional to conditionally add these functions here at this level? The function name is "optional" it is already deciding what should be shown for a condition.

Reply:
1. (useconstraints-onlineresource-validators kms-index)
           (collectioncitations-onlineresource-validators kms-index)
           (publicationreferences-onlineresource-validators kms-index) might not belong here since these MimeTypes don't belong to RelatedUrls.

2. DataCenter are validated twice, on different things. After merging, one would be lost. So They need to be combined in the end.

Reply:
A test needs to be added to introduce two errors. One is the DataCenter RelatedUrl/MimeType error, the other is the error from :DataCenters (match-kms-keywords-validation
                           kms-index :providers msg/data-center-not-matches-kms-keywords)})))
Make sure both are returned as warnings, none is lost.